### PR TITLE
`azurerm_kubernetes_*` - Upgrade to go-azure-sdk

### PR DESCRIPTION
**Pull request by aristosvo**
PR to migrate to go-azure-sdk for `azurerm_kubernetes_*`

**Status**: compiling,  run all tests, rebased

**Next steps**: Review time!

## Acceptance tests
Most passing, leftovers seem random failures except on `MaintenanceConfig` tests

```bash
=== CONT  TestAccKubernetesCluster_completeMaintenanceConfig
testcase.go:110: Step 1/2 error: Error running apply: exit status 1
Error: creating/updating maintenance config for Managed Cluster (Subscription: "*******"
Resource Group Name: "acctestRG-aks-221017225106165211"
Resource Name: "acctestaks221017225106165211"): maintenanceconfigurations.MaintenanceConfigurationsClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="UnmarshalError" Message="UnmarshalEntity encountered error: parsing time \"\\\"2021-12-30 12:00:00 +0000 UTC\\\"\" as \"\\\"2006-01-02T15:04:05Z07:00\\\"\": cannot parse \" 12:00:00 +0000 UTC\\\"\" as \"T\"."
with azurerm_kubernetes_cluster.test,
on terraform_plugin_test.tf line 11, in resource "azurerm_kubernetes_cluster" "test":
11: resource "azurerm_kubernetes_cluster" "test" {

=== CONT  TestAccKubernetesCluster_updateMaintenanceConfig
testcase.go:110: Step 3/6 error: Error running apply: exit status 1
Error: creating/updating Maintenance Configuration for Managed Kubernetes Cluster ("Managed Cluster (Subscription: \"*******\"\nResource Group Name: \"acctestRG-aks-221017225636884801\"\nResource Name: \"acctestaks221017225636884801\")"): maintenanceconfigurations.MaintenanceConfigurationsClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="UnmarshalError" Message="UnmarshalEntity encountered error: parsing time \"\\\"2021-12-30 12:00:00 +0000 UTC\\\"\" as \"\\\"2006-01-02T15:04:05Z07:00\\\"\": cannot parse \" 12:00:00 +0000 UTC\\\"\" as \"T\"."
with azurerm_kubernetes_cluster.test,
on terraform_plugin_test.tf line 11, in resource "azurerm_kubernetes_cluster" "test":
11: resource "azurerm_kubernetes_cluster" "test" {
--- FAIL: TestAccKubernetesCluster_updateMaintenanceConfig (793.81s)
```

And fixed:

```bash
 make acctests SERVICE='containers' TESTARGS='-run=TestAccKubernetesCluster_completeMaintenanceConfig\|TestAccKubernetesCluster_updateMaintenanceConfig'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/containers -run=TestAccKubernetesCluster_completeMaintenanceConfig\|TestAccKubernetesCluster_updateMaintenanceConfig -timeout 180m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccKubernetesCluster_completeMaintenanceConfig
=== PAUSE TestAccKubernetesCluster_completeMaintenanceConfig
=== RUN   TestAccKubernetesCluster_updateMaintenanceConfig
=== PAUSE TestAccKubernetesCluster_updateMaintenanceConfig
=== CONT  TestAccKubernetesCluster_completeMaintenanceConfig
=== CONT  TestAccKubernetesCluster_updateMaintenanceConfig
--- PASS: TestAccKubernetesCluster_completeMaintenanceConfig (638.73s)
--- PASS: TestAccKubernetesCluster_updateMaintenanceConfig (774.75s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/containers    776.536s
```

and a different error showing up:
```bash
=== CONT  TestAccKubernetesCluster_upgradeSettings
------- Stderr: -------
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4549de4]
goroutine 335 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.ConvertDefaultNodePoolToAgentPool(0xc003363710)
  /opt/teamcity-agent/work/5e6516bb4d10eb66/internal/services/containers/kubernetes_nodepool.go:702 +0x9a4
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.resourceKubernetesClusterCreate(0x0?, {0x5b16880?, 0xc001e13000}) 
```

And fixed:
```bash
> make acctests SERVICE='containers' TESTARGS='-run=TestAccKubernetesCluster_upgradeSettings'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/containers -run=TestAccKubernetesCluster_upgradeSettings -timeout 180m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccKubernetesCluster_upgradeSettings
=== PAUSE TestAccKubernetesCluster_upgradeSettings
=== CONT  TestAccKubernetesCluster_upgradeSettings
--- PASS: TestAccKubernetesCluster_upgradeSettings (889.38s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/containers    891.108s
```

### CODE REVIEW

File: internal/services/containers/kubernetes_cluster_resource.go
Comment (aristosvo) [inline internal/services/containers/kubernetes_cluster_resource.go:914]: @tombuildsstuff `ApiServerAccessProfile` is not returned by default anymore, which means I'm setting values by default now. Does this require this  change require a state migration?

Or is there a simpler solution?

File: internal/services/containers/kubernetes_cluster_resource.go
Comment (aristosvo) [inline internal/services/containers/kubernetes_cluster_resource.go:915]: @tombuildsstuff `ApiServerAccessProfile` is not returned when having 'default' settings anymore, which means I'm setting values by default now below. Does this mean I should add `Default: true` to the schema? Does this change requires a state migration?

Or is there a simpler solution?

**Comment by djsly** [comment]
@aristosvo what's the best practice, should I wait till you merge ? or should I create a new PR based on your PR to add missing features that the SDK brings in ?

**Comment by aristosvo** [comment]
I think HC wants to merge this one first anyway (and it might still need some small changes to comply).

But that shouldn't stop you from working on it. I'm not sure how good your rebase skills are, but a PR based on this work should be possible with a relatively simple rebase after merging.

File: internal/services/containers/kubernetes_cluster_resource.go
Comment (tombuildsstuff) [reply internal/services/containers/kubernetes_cluster_resource.go:915]: Taking a look through here this should be fine (but unnecessary) since this implicitly defaults to `false` in the read function?

**Review by stephybun — COMMENTED** [review]
The tests look mostly fine, there is one new failure which looks like it might have coincided with the dalek running. We triggered another round of tests, once those are done and the merge conflict is resolved this should be good to go!

File: internal/services/containers/kubernetes_cluster_resource.go
Comment (aristosvo) [reply internal/services/containers/kubernetes_cluster_resource.go:915]: Indeed, it implicitly defaults to 'false' in the read function, which is a change. I made it here explicit.

The main question is whether this defaulting (implicit or explicit) is a breaking change which requires a state migration, as this is also a `ForceNew` setting.

**Review by stephybun — APPROVED** [review]
Thanks @aristosvo LGTM 🥮

**Pull request merged**
